tests/thread_test.c

```diff
@@ -47,6 +50,12 @@ static int s_test_thread_creation_join_fn(struct aws_allocator *allocator, void
         aws_thread_get_detach_state(&thread),
         "thread state should have returned JOIN_COMPLETED");
 
+#ifdef AWS_PTHREAD_HAS_GETNAME
+    ASSERT_CURSOR_VALUE_STRING_EQUALS(
+        aws_byte_cursor_from_c_str("MyThreadName"), test_data.thread_name, "thread name equals");
+#endif
```

Comment:
Instead of using the ifdef, you could just call get_name(), and assert that either:
- error is PLATFORM_NOT_SUPPORTED
- OR name is the same one we passed in